testutils/src/main/scala/za/co/absa/enceladus/testutils/infoFileComparison/AtumModelUtils.scala

```diff
@@ -0,0 +1,196 @@
+    * ControlMeasure's class extension adding compareWith
+    * @param controlMeasure Control Measure instance
+    */
+  implicit class ControlMeasureOps (controlMeasure: ControlMeasure){
+    /**
+      * Compare this ControLMeasure with the one passed in
+      * @param otherControlMeasure Second control measure
+      * @return Returns a list of model differences
+      */
+    def compareWith(otherControlMeasure: ControlMeasure): List[ModelDifference[_]] = {
+      val metadataDifferences = controlMeasure.metadata.compareWith(otherControlMeasure.metadata, "metadata")
+      val checkpointsDifferences = controlMeasure.checkpoints.zipWithIndex.foldLeft(List[ModelDifference[_]]()) {
+        case (agg, (value, index)) =>
+          value.compareWith(otherControlMeasure.checkpoints(index), s"checkpoints[$index]") ::: agg
+      }
+      metadataDifferences ::: checkpointsDifferences
+    }
+  }
+
+  /**
+    * ControlMeasureMetadata's class extension adding compareWith
+    * @param metadata ControlMeasureMetadata instance
+    */
+  implicit class ControlMeasureMetadataOps (metadata: ControlMeasureMetadata){
+    /**
+      * Compare this ControlMeasureMetadata with the one passed in
+      * @param otherMetadata Second control measure metadata
+      * @param curPath Path to the ControlMeasureMetadata through the model
+      * @return Returns a list of model differences
+      */
+    def compareWith(otherMetadata: ControlMeasureMetadata, curPath: String): List[ModelDifference[_]] = {
+      val diffs = simpleCompare(metadata.sourceApplication,
+        otherMetadata.sourceApplication, s"$curPath.sourceApplication") ::
+      simpleCompare(metadata.country, otherMetadata.country, s"$curPath.country") ::
+      simpleCompare(metadata.historyType, otherMetadata.historyType, s"$curPath.historyType") ::
+      simpleCompare(metadata.dataFilename, otherMetadata.dataFilename, s"$curPath.dataFilename") ::
+      simpleCompare(metadata.sourceType, otherMetadata.sourceType, s"$curPath.sourceType") ::
+      simpleCompare(metadata.version, otherMetadata.version, s"$curPath.version") ::
+      simpleCompare(metadata.informationDate, otherMetadata.informationDate, s"$curPath.informationDate") ::
+        Nil
+
+      val additionalInfoDiff = additionalInfoComparison(metadata.additionalInfo,
+        otherMetadata.additionalInfo,
+        s"$curPath.additionalInfo")
+
+      (diffs flatten) ::: additionalInfoDiff
+    }
+
+    /**
+      * Compare additional information from ControlMeasureMetadata
+      * @param was     Value it had in reference
+      * @param is Value it has now
+      * @param curPath Path to the AdditionalInfo through the model
+      * @return Returns a list of model differences
+      */
+    private def additionalInfoComparison(was: Map[String, String],
+                                         is: Map[String,String],
+                                         curPath: String): List[ModelDifference[_]] = {
+      was.flatMap {
```

Comment:
Because `was` is a Map it can be a hash map. So the order of elements can differ depending on various circumstances. So from time to time, the list of differences can appear in a different order.

Consider either
* Convert it to a List first, sort it, and then apply `flatMap` or
* Sort the final list of `ModelDifference` instances before returning.